### PR TITLE
Use listener scoped sasl mechanisms

### PR DESCRIPTION
**Pull request by danielpetisme**
### Description 

This PR aims at closing #378 

### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

https://confluentinc.atlassian.net/browse/DOCS-8661
https://confluentinc.atlassian.net/browse/DOCS-8662

### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._

### CODE REVIEW

**Review by ybyzek — COMMENTED** [review]
@danielpetisme thank you for identifying the issue and proactively submitting a PR.  I validated a run of `cp-demo` with this change.  But since I think the question here is best practice: copying some folks from the Slack thread to validate the change to `cp-demo`.  @rajinisivaram, @soondenana, or @omkreddy if one of you could kindly confirm the change.

**Comment by ybyzek** [comment]
@danielpetisme update: while previously I validated just running cp-demo's start script, I just looked through usage of the users mentioned in the jaas config file and there are references to them in a few places in `cp-demo`.  If they are being removed from `scripts/security/broker_jaas.conf` then these other references need to be investigated and take corrective action validated/cleaned/removed.

### `badclient`

```
[7:45:38] ~/git/cp-demo(PR-379) ✗: grep -ir badclient-secret *
scripts/security/badclient_without_interceptors.config:    password=\"badclient-secret\";

[7:45:43] ~/git/cp-demo(PR-379) ✗: grep -ir badclient_without_interceptors.config *
[7:47:03] ~/git/cp-demo(PR-379) ✗: 
```

Suggested action: also delete the file `scripts/security/badclient_without_interceptors.config`

### `client`

```
[7:49:21] ~/git/cp-demo(PR-379) ✗: grep -ir client-secret *
scripts/security/badclient_without_interceptors.config:    password=\"badclient-secret\";
scripts/security/client_without_interceptors.config:    password=\"client-secret\"; <<<<
scripts/security/unknownclient_without_interceptors.config:    password=\"unknownclient-secret\";

[7:49:25] ~/git/cp-demo(PR-379) ✗: grep -ir client_without_interceptors.config *
scripts/consumers/listen_wikipedia.parsed.replica.sh:ARGS="--bootstrap-server kafka1:9091 --consumer.config /etc/kafka/secrets/client_without_interceptors.config"
scripts/consumers/listen_wikipedia.parsed.count-by-channel.sh:ARGS="--bootstrap-server kafka1:9091 --consumer.config /etc/kafka/secrets/client_without_interceptors.config"
scripts/consumers/listen_wikipedia.failed.sh:  --consumer.config /etc/kafka/secrets/client_without_interceptors.config
scripts/consumers/listen_wikipedia.parsed.sh:ARGS="--bootstrap-server kafka1:9091 --consumer.config /etc/kafka/secrets/client_without_interceptors.config"
scripts/consumers/listen_users.sh:ARGS="--bootstrap-server kafka1:9091 --consumer.config /etc/kafka/secrets/client_without_interceptors.config"
```

**Review by mikebin — APPROVED** [review]
LGTM

**Comment by danielpetisme** [comment]
Thanks, @ybyzek for the double-check.

The following files seem not to be used so it should not be a problem.
* `badclient_without_interceptors.config`
* `unknownclient_without_interceptors.config`

Regarding `client_without_interceptors.config` I'll have a close look at it.

**Review by omkreddy — APPROVED** [review]
LGTM. I agree, we can recommend using explicit listener scoped sasl mechanism.

**Comment by danielpetisme** [comment]
I gave a look to the consumer using `client_without_interceptors.config`

TL;DR: none of the listed file actually work in the current `6.2.0-post` branch. We can either remove or fix them but I would recommend doing so in another PR.

* `listen_users.sh` uses `client_without_interceptors.config` and tries to read from the `users` topics. The only mentions of this script is  in the [onprem.rst](https://github.com/confluentinc/cp-demo/blob/6.2.0-post/docs/on-prem.rst) doc references `client_sasl_plain.config`. By changing the config file to `client_sasl_plain.config` the client works as expected. However, since the script is never used I don't know if it's better to fix it or remove it.

* `listen_wikipedia.parsed.count-by-channel.sh`, `listen_wikipedia.parsed.replica.sh`, `isten_wikipedia.failed.sh` and `listen_wikipedia.parsed.sh` all this scripts use a Kafka Avro console consumer but does not provide credentials to connect to  the schema Registry, so I assume they are not used anymore and would make more sense to simply remove them.

WDYT?

**Comment by ybyzek** [comment]
@danielpetisme thank you for looking into this.  I double-checked and also cannot find reference to any of the bash "listen" scripts you mentioned.  Let's nuke them.

(history: I think these scripts were useful ~3 years ago, before cp-demo had RBAC and before it had HTTPS endpoints, and when C3 didn't have great topic inspection capabilities.  So at that time, it was nice to have a handy bash script to see the messages.  Since then, cp-demo changed and C3 got really great abilities to read messages in Kafka topics, so the bash scripts aren't used any more).

**Comment by danielpetisme** [comment]
Ok I'll remove them.
Thank for the double check!

Le lun. 19 juil. 2021 à 19:49, Yeva Byzek ***@***.***> a
écrit :

**Comment by danielpetisme** [comment]
Cleaning done.
@ybyzek do you see other stuff blocking the merge?
Thanks

**Review by ybyzek — APPROVED** [review]
@danielpetisme thank you for everything -- identifying the issue, identifying the fix, and submitting the PR.  LGTM
